api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/ChildDateOfBirthCalculator.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.dhsc.htbhf.claimant.message.processor;
+
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import uk.gov.dhsc.htbhf.claimant.entitlement.CycleEntitlementCalculator;
+import uk.gov.dhsc.htbhf.claimant.entity.PaymentCycle;
+
+import java.time.LocalDate;
+
+/**
+ * Contains utility methods for helping to figure out information pertaining to the dates of birth
+ * for the Claimant.
+ */
+@Component
+public class ChildDateOfBirthCalculator {
+
+    private final CycleEntitlementCalculator cycleEntitlementCalculator;
+
+    public ChildDateOfBirthCalculator(CycleEntitlementCalculator cycleEntitlementCalculator) {
+        this.cycleEntitlementCalculator = cycleEntitlementCalculator;
+    }
+
+    /**
+     * Calculates how many of the children's dates of birth listed on the given PaymentCycle have birthdays
+     * which would affect the next payment after this cycle. The dates that would affect the next payment
+     * are between the last entitlement date of the current cycle and the final entitlement date of the next
+     * cycle. Those with birthdays on the start boundary date (the date of the final entitlement date of
+     * the current cycle) are not included, but those on the end boundary date (the final entitlement date
+     * of the next cycle) are included.
+     *
+     * @param paymentCycle The current PaymentCycle including the relevant dates of birth of the children.
+     * @return The number of children who will soon turn 4 that affect the next Payment.
+     */
+    public int getNumberOfChildrenTurningFourWhichAffectsNextPayment(PaymentCycle paymentCycle) {
```

Comment:
I've been trying to come up with a more succinct name. All I've managed so far is `getNumberOfChildrenTurningFourAffectingNextPayment`

Reply:
Yep, that's better, I've updated.